pkg/testidentification/ocp_variants_test.go

```diff
@@ -85,6 +85,10 @@ func Test_openshiftVariants_IdentifyVariants(t *testing.T) {
 			clusterData: models.ClusterData{Release: "4.13", Network: "sdn", Platform: "azure", Architecture: "arm64", Topology: "single"},
 			want:        []string{"azure", "arm64", "sdn", "ha"},
 		},
+		{
+			name:        "periodic-ci-openshift-release-master-nightly-4.14-e2e-agent-ha-dualstack",
```

Comment:
Probably the compact would be a better choice for that?

Reply:
This test mainly just checks the variant name `agent` so I think this is enough, others are default.

Reply:
Isn't fetching the results from that kind of job?

Reply:
No it's just a small unit test to make sure we saw "agent" in the job name, no bearing on what we fetch.

Reply:
@andfasano ok, so I see there is a bug in the existing code which encounters when we use the job name  `periodic-ci-openshift-release-master-nightly-4.14-e2e-agent-compact-ipv4` in the test case which results adding the topology twice in the variant array.

Reply:
Lets keep the release check too.